File: single-language-analysis/dataset-files/java/TestSmell029.java
// Original URL: https://github.com/muenchhausen/3wins/blob/57b86a71a57b9a62f811d7d568b11d9d676821fd/GameTest.java#L190-L206

public class TestSmell029 {
    @Test
    public void check_winner_should_return_GREEN_if_3_GREEN_are_on_the_top()
    {
        Game game = new Game(n3winAlgorithm,CALCULATION_DEPTH);
        Chip field[][] = new Chip[5][5];
        
        game.placeChipInColumn(field, 2,Chip.GREEN);
        game.placeChipInColumn(field, 2,Chip.RED);
        game.placeChipInColumn(field, 2,Chip.GREEN);
        game.placeChipInColumn(field, 2,Chip.GREEN);
        game.placeChipInColumn(field, 2,Chip.GREEN);
        Chip result= n3winAlgorithm.theWinnerIs(field);
        assertEquals(Chip.GREEN,result);

        result = threeWinAlgorithm.theWinnerIs(field);
        assertEquals(Chip.GREEN,result);
    }
}

File: single-language-analysis/dataset-files/python/test_smell_029.py
import pytest

class TestGame:
    def test_smell_029(self):
        game = Game(n3win_algorithm, CALCULATION_DEPTH)
        field = [
            [None, None, None, None, None],
            [None, None, None, None, None],
            [None, None, None, None, None],
            [None, None, None, None, None],
            [None, None, None, None, None],
        ]
        
        game.place_chip_in_column(field, 2, Chip.GREEN)
        game.place_chip_in_column(field, 2, Chip.RED)
        game.place_chip_in_column(field, 2, Chip.GREEN)
        game.place_chip_in_column(field, 2, Chip.GREEN)
        game.place_chip_in_column(field, 2, Chip.GREEN)
        
        result = n3win_algorithm.the_winner_is(field)
        assert result == Chip.GREEN

        result = three_win_algorithm.the_winner_is(field)
        assert result == Chip.GREEN